src/java/org/apache/cassandra/cql3/statements/BatchStatement.java

```diff
@@ -441,7 +445,15 @@ public ResultMessage execute(QueryState queryState, BatchQueryOptions options, l
             executeWithoutConditions(getMutations(queryState, options, false, timestamp, nowInSeconds, queryStartNanoTime),
                                      queryState, cl, queryStartNanoTime);
 
-        return new ResultMessage.Void();
+        ResultMessage<ResultMessage.Void> result = new ResultMessage.Void();
+        RequestSensors sensors = RequestTracker.instance.get();
+        for (ModificationStatement statement : statements)
```

Comment:
If all statements refer to the same context, are we then adding the same sensor over and over again?